client/state/current-user/actions.js

```diff
@@ -19,6 +20,29 @@ export function setCurrentUser( user ) {
 	};
 }
 
+export function fetchCurrentUser() {
+	return async ( dispatch ) => {
+		dispatch( {
+			type: CURRENT_USER_FETCH,
+		} );
+
+		try {
+			const user = await wpcom.me().get( {
+				meta: 'flags',
+			} );
+			const userData = filterUserObject( user );
+
+			const storedUserId = getStoredUserId();
+			if ( storedUserId != null && storedUserId !== userData.ID ) {
+				await clearStore();
+			}
+
+			setStoredUserId( userData.ID );
+			dispatch( setCurrentUser( userData ) );
+		} catch {}
```

Comment:
Should we at least log something here rather than silently failing?

Reply:
In the legacy implementations we have several `debug` calls and a `console.error`:

* https://github.com/Automattic/wp-calypso/blob/trunk/client/lib/user/user.js#L132
* https://github.com/Automattic/wp-calypso/blob/trunk/client/lib/user/user.js#L150
* https://github.com/Automattic/wp-calypso/blob/trunk/client/lib/user/user.js#L155

I really don't see much value in those, especially for the end user, which is why I haven't transferred them to the Redux implementation. I'm happy to reconsider if anyone disagrees 😉